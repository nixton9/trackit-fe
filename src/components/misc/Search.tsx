import React, { useState, useEffect, useRef } from 'react'
import { useToggleElement } from '../../utils/useToggleElement'
import { Styled } from '../../styles/Search.styles'
import { ReactComponent as GlassIcon } from '../../assets/icons/glass.svg'
import { useHistory } from 'react-router-dom'

const Search: React.FC = () => {
  const [inputVal, setInputVal] = useState('')
  const [open, setOpen, overlayEl] = useToggleElement()

  const inputRef: React.RefObject<HTMLInputElement> = useRef(null)

  const history = useHistory()

  useEffect(() => {
    if (!open) setInputVal('')
  }, [open])

  const handleIconClick = () => {
    if (open) {
      handleSubmit()
    } else {
      setOpen(true)
      setTimeout(
        () => inputRef && inputRef.current && inputRef.current.focus(),
        400
      )
    }
  }

  const handleSubmit = (e?: React.FormEvent<HTMLFormElement>) => {
    e && e.preventDefault()
    if (inputVal.length) {
      history.push(`/search/${inputVal}`)
      setOpen(false)
      setInputVal('')
    }
  }

  return (
    <>
      <Styled.SearchIcon open={open} className="mbl-click">
        <GlassIcon onClick={handleIconClick} data-test-id="search-icon" />
        <form onSubmit={handleSubmit}>
          <Styled.SearchInput
            ref={inputRef}
            open={open}
            type="text"
            placeholder="Search for a note, task, habit or expense"
            value={inputVal}
            onChange={e => setInputVal(e.target.value)}
            data-test-id="search-input"
          />
        </form>
      </Styled.SearchIcon>
      <Styled.SearchOverlay open={open} ref={overlayEl} />
    </>
  )
}

export default Search
