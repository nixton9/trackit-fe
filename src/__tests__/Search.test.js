import React from 'react'
import { render } from '@testing-library/react'
import Search from '../components/misc/Search'

describe('Search', () => {
  it('matches snapshot', () => {
    const { container } = render(<Search />)

    expect(container).toMatchSnapshot()
  })

  it('has the input', () => {
    const { getByPlaceholderText } = render(<Search />)

    expect(
      getByPlaceholderText('Search for a note, task, habit or expense')
    ).toBeInTheDocument()
  })
})
