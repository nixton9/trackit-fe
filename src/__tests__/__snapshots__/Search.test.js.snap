// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search matches snapshot 1`] = `
<div>
  <div
    class="sc-AxjAm hXhgNM"
  >
    <svg>
      glass.svg
    </svg>
    <form>
      <input
        class="sc-AxirZ ioKDud"
        placeholder="Search for a note, task, habit or expense"
        type="text"
        value=""
      />
    </form>
  </div>
  <div
    class="sc-AxiKw bUCRFN"
  />
</div>
`;
